srtcore/core.cpp

```diff
@@ -3297,18 +3298,16 @@ void srt::CUDT::synchronizeWithGroup(CUDTGroup* gp)
         m_pRcvBuffer->applyGroupTime(rcv_buffer_time_base, rcv_buffer_wrap_period, m_iTsbPdDelay_ms * 1000, rcv_buffer_udrift);
         leaveCS(m_RecvLock);
 
+        updated_tsbpd = true;
```

Comment:
Might be cleaner to set peer rexmit flag right after the `applyGroupTime(..)` instead. Not 100% clean, but at least more obvious.
```
m_pRcvBuffer->setPeerRexmitFlag(m_bPeerRexmitFlag);
```

Reply:
Do you mean keep `updateSrtRcvSettings()` inside `else`?
I move `updateSrtRcvSettings()` out of `else` mainly because more settings may be updated in `updateSrtRcvSettings()` and it's easy to forget to also update these settings here, just as the bug I'm fixing.

Reply:
Or we can delay the creation of `m_pRcvBuffer`?

Reply:
Delaying the creation of the receiver buffer was my first attempt, but it pulls in a lot of changes, unfortunately. Can't do this close to release candidate.

Here I meant leaving most of the code as is, just adding `setPeerRexmitFlag` right after the `applyGroupTime`.

Reply:
```patch
diff --git a/srtcore/core.cpp b/srtcore/core.cpp
index e9f333d3..6351a8d3 100644
--- a/srtcore/core.cpp
+++ b/srtcore/core.cpp
@@ -3295,6 +3295,9 @@ void srt::CUDT::synchronizeWithGroup(CUDTGroup* gp)
         // time to not fill a network window.
         enterCS(m_RecvLock);
         m_pRcvBuffer->applyGroupTime(rcv_buffer_time_base, rcv_buffer_wrap_period, m_iTsbPdDelay_ms * 1000, rcv_buffer_udrift);
+#if ENABLE_NEW_RCVBUFFER
+        m_pRcvBuffer->setPeerRexmitFlag(m_bPeerRexmitFlag);
+#endif
         leaveCS(m_RecvLock);
 
         HLOGF(gmlog.Debug,  "AFTER HS: Set Rcv TsbPd mode: delay=%u.%03us GROUP TIME BASE: %s%s",

```

Reply:
> Delaying the creation of the receiver buffer was my first attempt, but it pulls in a lot of changes, unfortunately. Can't do this close to release candidate.

Great! Looking forward to your refactor :)